platform/linux-generic/test/validation/api/ml/ml_linux.c

```diff
@@ -0,0 +1,1259 @@
+	global.data.input_seg = &global.input_seg;
+	global.input_seg.size = sizeof(int32_t);
+	global.input_seg.addr = &global.x;
+
+	global.data.num_output_seg = NUM_OUTPUTS;
+	global.data.output_seg = &global.output_seg;
+	global.output_seg.size = sizeof(int32_t);
+	global.output_seg.addr = &global.y;
+
+	if (fill_model_param("simple_linear.onnx", &global.model_param))
+		return -1;
+
+	/* Create ML model */
+	global.ml_model = odp_ml_model_create(MODEL_NAME, &global.model_param);
+	if (global.ml_model == ODP_ML_MODEL_INVALID) {
+		ODPH_ERR("Create ML model failed\n");
+		free(global.model_param.model);
+		return -1;
+	}
+
+	/* Asynchronous mode with event completion is not supported */
+	if (!((ml_capa->load.compl_mode_mask & ODP_ML_COMPL_MODE_EVENT) ||
+	      (ml_capa->run.compl_mode_mask & ODP_ML_COMPL_MODE_EVENT)))
+		return 0;
+
+	/* Create a queue for sending ML completion event to */
+	odp_queue_param_init(&queue_param);
+	queue_param.type        = ODP_QUEUE_TYPE_SCHED;
+	queue_param.sched.sync  = ODP_SCHED_SYNC_PARALLEL;
+	queue_param.sched.prio  = odp_schedule_default_prio();
+	queue_param.sched.group = ODP_SCHED_GROUP_ALL;
+
+	global.queue = odp_queue_create("ML compl queue", &queue_param);
+	if (global.queue == ODP_QUEUE_INVALID) {
+		ODPH_ERR("Queue create failed\n");
+		free(global.model_param.model);
+		return -1;
+	}
+
+	/* Create an ML job completion pool */
+	if (ml_capa->pool.max_num < NUM_COMPL) {
+		ODPH_ERR("Too small ML compl pool %u\n", ml_capa->pool.max_num);
+		return -1;
+	}
+
+	odp_ml_compl_pool_param_init(&ml_pool_param);
+	ml_pool_param.num = NUM_COMPL;
+
+	global.compl_pool = odp_ml_compl_pool_create(COMPL_POOL_NAME, &ml_pool_param);
+	if (global.compl_pool == ODP_POOL_INVALID) {
+		free(global.model_param.model);
+		ODPH_ERR("Create ML completion pool failed\n");
+		return -1;
+	}
+
+	return 0;
+}
+
+static int ml_suit_term(void)
+{
+	if (global.compl_pool != ODP_POOL_INVALID &&
+	    odp_pool_destroy(global.compl_pool)) {
+		ODPH_ERR("Completion pool destroy failed\n");
+		return -1;
+	}
+
+	if (global.ml_model && odp_ml_model_destroy(global.ml_model)) {
+		ODPH_ERR("Destroy ML model failed\n");
+		return -1;
+	}
+
+	if (global.queue != ODP_QUEUE_INVALID &&
+	    odp_queue_destroy(global.queue)) {
+		ODPH_ERR("Destroy ML queue failed\n");
+		return -1;
+	}
+
+	free(global.model_param.model);
+
+	return 0;
+}
+
+static int check_ml_support(void)
+{
+	if (global.disabled)
+		return ODP_TEST_INACTIVE;
+
+	return ODP_TEST_ACTIVE;
+}
+
+static int check_load_sync(void)
+{
+	if (global.disabled)
+		return ODP_TEST_INACTIVE;
+
+	if (global.ml_config.load_mode_mask & ODP_ML_COMPL_MODE_SYNC)
+		return ODP_TEST_ACTIVE;
+
+	return ODP_TEST_INACTIVE;
+}
+
+static int check_load_poll_sync(void)
+{
+	if (global.disabled)
+		return ODP_TEST_INACTIVE;
+
+	if ((global.ml_config.load_mode_mask & ODP_ML_COMPL_MODE_POLL) &&
+	    (global.ml_config.load_mode_mask & ODP_ML_COMPL_MODE_SYNC))
+		return ODP_TEST_ACTIVE;
```

Comment:
If sync mode (load/unload) is not supported, poll or event mode load/unload is not tested. I.e. then all these load/unload tests are skipped.

It would be better to test both load and unload in the same mode, and have only three test cases: sync load/unload, poll load/unload, event load/unload.

Reply:
Done. Separate unload_async tests removed.